upup/pkg/fi/cloudup/awsup/elbv2_targetgroups.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awsup
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"k8s.io/klog/v2"
+)
+
+type TargetGroupInfo struct {
+	TargetGroup *elbv2.TargetGroup
+	Tags        []*elbv2.Tag
+}
+
+// ARN returns the ARN of the load balancer.
+func (i *TargetGroupInfo) ARN() string {
+	return aws.StringValue(i.TargetGroup.TargetGroupArn)
```

Comment:
perhaps return an empty string if `i.TargetGroup` is nil rather than panic

Reply:
aws.StringValue does this :-)  It's the equivalent of our fi.ValueOf (and previously fi.StringValue).  I try to use the aws functions when dealing with aws inputs, and the fi functions when using with our own, though the function definitions are identical (modulo generics)

Reply:
Accessing TargetGroupArn if TargetGroup is nil will panic before even entering the StringValue function though, right?

Reply:
Doh, you're right, sorry!